From bf65791d529ef332fa2bc3e3e26252ad84e00d1b Mon Sep 17 00:00:00 2001
From: Arthur Williams <taaparthur@gmail.com>
Date: Sun, 20 Jun 2021 01:21:26 -0500
Subject: [PATCH 2/2] Faster conflict detection

Replace the use of grep to determine the unique dependencies with
sort/uniq.
---
 kiss | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kiss b/kiss
index c0d29f2..a73aca5 100755
--- a/kiss
+++ b/kiss
@@ -1122,7 +1122,7 @@ pkg_conflicts() {
     # Store the list of found conflicts in a file as we'll be using the
     # information multiple times. Storing things in the cache dir allows
     # us to be lazy as they'll be automatically removed on script end.
-    grep -Fxf "$_tmp_file_pre" -- "$@" 2>/dev/null > "$_tmp_file" || :
+    sed '/\/$/d' "$@" | sort "$_tmp_file_pre" - | uniq -d > "$_tmp_file" ||:
 
     # Enable alternatives automatically if it is safe to do so.
     # This checks to see that the package that is about to be installed
@@ -1151,7 +1151,7 @@ pkg_conflicts() {
         # this work.
         #
         # Pretty nifty huh?
-        while IFS=: read -r _ con; do
+        while read -r con; do
             printf '%s\n' "Found conflict $con"
 
             # Create the "choices" directory inside of the tarball.
-- 
2.33.0

